pymongo/collection.py

```diff
@@ -2277,12 +2277,9 @@ def _find_and_modify(session, sock_info, retryable_write):
                         'writes.')
                 cmd["arrayFilters"] = array_filters
             if hint is not None:
-                if sock_info.max_wire_version < 8:
-                    raise ConfigurationError(
-                        'Must be connected to MongoDB 4.2+ to use hint.')
-                if not write_concern.acknowledged:
+                if sock_info.max_wire_version < 9:
```

Comment:
A strict reading of the crud spec and tests means this needs to be:
```
if sock_info.max_wire_version < 8 or (not acknowledged && sock_info.max_wire_version < 9): raise <>
```
See:
```
class FindOneAndDeleteOptions {
...
  /**
   * The index to use. Specify either the index name as a string or the index key pattern.
   * If specified, then the query system will only consider plans using the hinted index.
   *
   * This option is sent only if the caller explicitly provides a value. The default is to not send a value.
   * This option is only supported by servers >= 4.4. Older servers >= 4.2 will report an error for using this option.
   * For servers < 4.2, the driver MUST raise an error if the caller explicitly provides a value.
   * For unacknowledged writes and servers < 4.4, the driver MUST raise an error if the caller explicitly provides a value.
   *
   * @see https://docs.mongodb.com/manual/reference/command/findAndModify/
   */
  hint: Optional<(String | Document)>;
```
https://github.com/mongodb/specifications/blob/master/source/crud/crud.rst#find-and-modify

Reply:
Fixed